public/app/features/expressions/components/Threshold.tsx

```diff
@@ -0,0 +1,148 @@
+import { css } from '@emotion/css';
+import React, { FC, FormEvent } from 'react';
+
+import { GrafanaTheme2, SelectableValue } from '@grafana/data';
+import { ButtonSelect, InlineField, InlineFieldRow, Input, Select, useStyles2 } from '@grafana/ui';
+import { EvalFunction } from 'app/features/alerting/state/alertDef';
+
+import { ClassicCondition, ExpressionQuery, thresholdFunctions } from '../types';
+
+interface Props {
+  labelWidth: number;
+  refIds: Array<SelectableValue<string>>;
+  query: ExpressionQuery;
+  onChange: (query: ExpressionQuery) => void;
+}
+
+const defaultThresholdFunction = EvalFunction.IsAbove;
+
+export const Threshold: FC<Props> = ({ labelWidth, onChange, refIds, query }) => {
+  const styles = useStyles2(getStyles);
+
+  const defaultEvaluator: ClassicCondition = {
+    type: 'query',
+    evaluator: {
+      type: defaultThresholdFunction,
+      params: [0, 0],
+    },
+    query: {
+      params: [],
+    },
+    reducer: {
+      params: [],
+      type: 'last',
+    },
+  };
+
+  const conditions = query.conditions ?? [defaultEvaluator];
```

Comment:
No need to check `query.conditions.length`? If the `query.conditions` is an empty array then, in the next line, we'll access index 0 of an empty array

Reply:
That's a good point I hadn't considered; I'll add a guard here to prevent that (and transform `[]` in to a `[defaultEvaluator]`.